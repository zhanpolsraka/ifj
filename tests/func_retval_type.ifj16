//4
class Main{
	static void run(){
		int a = retint();
	}
	static int retint(){
		return 1.5;
	}
}