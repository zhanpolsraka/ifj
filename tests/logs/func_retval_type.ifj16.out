---------------FILENAME: tests/func_retval_type.ifj16---------------


Instruction stack item [8] -> class with name "Main"
Instruction stack item [7] -> function with name "run"
Instruction stack item [6] -> instruction with operation 35 -> INSTR_CALL_FUNC
Instruction stack item [5] -> instruction with operation 28 -> ASSIGNMENT
Instruction stack item [4] -> end function
Instruction stack item [3] -> function with name "retint"
Instruction stack item [2] -> instruction with operation 37 -> INSTR_ASS_ARG
Instruction stack item [1] -> end function
Instruction stack item [0] -> end class


---------------MAKE RELATIONS--------------
                     *
                     *
                     *
------------------EXECUTE------------------

------------------OUTPUT-------------------

STEP 1 - non-func instructions
Found Main
STEP 1 - non-func instructions
Got in function - ignore instructions until I get out
Found run()
STEP 1 - non-func instructions
STEP 1 - non-func instructions
STEP 1 - non-func instructions
Got out of function
Got out of run()
STEP 1 - non-func instructions
Got in function - ignore instructions until I get out
STEP 1 - non-func instructions
STEP 1 - non-func instructions
Got out of function
STEP 1 - non-func instructions
Got out of Main
STEP 2 - validating existance of run() in Main
STEP 3 - executing run() instructions
INSTR_CALL_FUNC
INSERT 1.5
---------------------------------------------------
N0: type is 23, values is 1.5
---------------------------------------------------
---------------------------------------------------
N0: type is 23, values is 1.5
---------------------------------------------------
ASSIGN 1.500000 to 0x8bd4e0
---------------------------------------------------
N0: type is 23, values is 1.5
---------------------------------------------------
Assignment between 22 and 23
---------------------------------------------------
---------------------------------------------------

-----------------END OUTPUT----------------


----------------ALL RIGHT!!!---------------

----------------SYMBOL TREE----------------

Class Main
	Function -> run
Lokal variables: 
	Variable -> a, type INTEGER, value [0]
	Function -> retint


--------------INSTRUCTION LIST-------------

Instruction stack item [8] -> class with name "Main"
Instruction stack item [7] -> function with name "run"
Instruction stack item [6] -> instruction with operation 35 -> INSTR_CALL_FUNC
Instruction stack item [5] -> instruction with operation 28 -> ASSIGNMENT
Instruction stack item [4] -> end function
Instruction stack item [3] -> function with name "retint"
Instruction stack item [2] -> instruction with operation 37 -> INSTR_ASS_ARG
Instruction stack item [1] -> end function
Instruction stack item [0] -> end class



